### ISSUE TITLE
Interface for dense elementwise vector operations

### ISSUE THREAD

**Issue by jpola** [clMathLibraries/clSPARSE#86]
I think it might be useful and trivial to add interface for element wise vector operations like subtract, add, multiply
i.e:
```c++
clsparseStatus cldenseSsubtract(clsparseVector* R, const clsparseVector* X, const clsparseVector* Y clsparseControl control)
```

All infrastrucure is present in ```elementwise_transform.hpp```. Just add the API.

**Comment by kknox** [comment]
This is a repeat of the comment I made in PR #126

@jpola I've reviewed this code, and agree that this functionality is easy to add and could provide value for end users. However, I think we may need to go back revisit the design of our header files. Our primary purpose (or scope) is providing an API for a sparse BLAS library, and in the course of development we have developed a significant share of dense arithmetic too, as a sparse library is actually a mix of both sparse and dense operations.

I'm considering refactoring clsparse.h into a set of header files, for the primary purpose of documentation and function organization. Could it make sense to add a clDense.h and a clSolver.h? Should we continue with a monolithic header file?

**Comment by jpola** [comment]
It is not a simple yes:no question for me :smile: 
I'm used to have one header file for such libraries. If I cached your idea correctly you want to have:
* clDense.h - with the blas1 functions
* clSolver.h - solvers and preconditioners
* clSPARSE.h - everything else

If so, it would be more reasonable for me to take a closer look on the clSPARSE and make more headers related to transformation between matrix types, input and output and others. Then the clSPARSE will just collect all of them.

Right now I don't see a reason to split the main header unless you give me a good example and convince me that this is better solution than one *yet not so big* file.

**Comment by jpola** [comment]
Closed by #126
